appland/popup/record.js

```diff
@@ -3,50 +3,63 @@ const recordButton = document.querySelector('#appmap-record');
 const recordButtonGraphic = document.querySelector('.appmap-record-button');
 const urlInput = document.querySelector("#appland-url");
 const header = document.querySelector('.header');
-const browser = (window.chrome || window.browser);
 
 const colorReady = '#da0030';
 const colorDisabled = '#350020';
 
 let ellipsisTimeoutId = -1;
 
-function saveApplandUrl(url) {
-  browser.storage.local.set({applandUrl: url});
+document.addEventListener('DOMContentLoaded', onLoad);
+header.addEventListener('click', onHeaderClick);
+recordButton.addEventListener('change', (e) => {
+  e.target.checked ? startRecording() : stopRecording()
+});
+
+urlInput.addEventListener('change', (e) => {
+  saveApplandUrl(e.target.value);
+});
+
+async function onHeaderClick() {
+  const url = await getApplandUrl();
+  if (isUrlValid(url)) {
+    window.open(url, '_blank');
+  }
+  else {
+    alert(`${url} is not a valid URL`);
+  }
 }
 
 function getTabUrl(callback) {
-  browser.tabs.query({active: true, currentWindow: true}, (tabs) => {
+  chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
```

Comment:
Does this restrict us to `chrome`?

Reply:
Nope. Firefox offers both `browser` and `chrome` objects, specifically for compatibility with Chromium-based browsers. The functions off the `browser` object use `Promise`s, those off the `chrome` object use callback functions.